DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -411,67 +413,72 @@ for Mesh in ${LowResMeshes} ${HighResMesh} ; do
 
 	for Map in ${Maps} ${MyelinMaps} SphericalDistortion ArealDistortion EdgeDistortion StrainJ StrainR ; do
 		log_Msg "Map: ${Map}"
-
-		for MapMap in ${MyelinMaps} ; do
+		
+		for ((index = 0; index < ${#MyelinMapsArray[@]}; ++index))
+		do
+			MapMap="${MyelinMapsArray[index]}"
 			log_Msg "MapMap: ${MapMap}"
-
+			
 			if [ ${MapMap} = ${Map} ] ; then
```

Comment:
For the record, this confusing structure for "only run this part on the myelin maps" should really be a loop over only myelin maps, above the `for Map in ...` loop (and the "run this part on only the first myelin map" shouldn't be inside the myelin map loop either), populating a new list of `_BC` map names to be used in the `for Map in ...` loop (see the modification of `$Map` at the end of the myelin map section).  The loops with `MapMap` at the end of this same `for Map in ...` loop should also be moved after the outer loop.

However, that should probably be a separate PR after we have the errors fixed.

Reply:
I agree that this legacy part is confusing. The `for MapMap in ${MyelinMaps} ; do` loop should be changed to `if [[ " ${MyelinMapsArray[*]} " =~ " ${Map} " ]]; then`, to only apply the bias field correct on myelin map when `${Map}` is an element in the array `${MyelinMapsArray}`.

Reply:
Also, I would not use `=~` to resolve this, trusting variables to not contain regexp-special characters is not a great idea.

Reply:
Okay, I see. So you are suggesting to divide the loop into two sequential loops, one for the myelin maps list (`${MyelinMaps}`), the following one for the rest of maps. This should make the code more readable.

Reply:
Correction: the myelin map BC script shouldn't be called if the list of myelin maps is empty, so that is a conditional it should be inside.  Other than that, it should only be inside the loop over resolutions, not inside any additional loop.

Given that the myelin map list is already expected to contain the value 'MyelinMap' if it contains it at all, the myelin map BC script should probably take that entire list as-is if it handles them internally (which would be my suggestion, to avoid needing a myelin-specific loop in DeDrift...).

Reply:
Yes, specifically, a loop to do the myelin-only steps first, and then a loop over the other maps AND the _BC myelin maps, to do the resampling that is common to all of them.

Reply:
The latest commit should address the comments. I would rather not add the `addtional-myelin-map` part into the `MyelinMap_BC.sh`.